### PR TITLE
Fix being able to move control nodes in containers with arrow keys.

### LINKED ISSUE #55260: It's possible to move Control node inside container if using arrow keys
### Godot version

3.4 / 4.0

### System information

W10

### Issue description

Normally when you try to move a Control node inside Container, you get a message that node is inside container and you can't. 4.0 even blocks inspector properties.

But you can still use arrow keys:
ebNTKErYY4

### Steps to reproduce

1. Add any Container node to scene
2. Add any Control node as its child
3. Use arrow keys to move the control

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by LlamaLad7**
Fixes #55260

### CODE REVIEW

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 1]
```diff
@@ -2097,8 +2097,16 @@ bool CanvasItemEditor::_gui_input_move(const Ref<InputEvent> &p_event) {
 	if (k.is_valid() && k->is_pressed() && (tool == TOOL_SELECT || tool == TOOL_MOVE) &&
 			(k->get_keycode() == Key::UP || k->get_keycode() == Key::DOWN || k->get_keycode() == Key::LEFT || k->get_keycode() == Key::RIGHT)) {
 		if (!k->is_echo()) {
-			// Start moving the canvas items with the keyboard
-			drag_selection = _get_edited_canvas_items();
+			// Start moving the canvas items with the keyboard, if they are movable
+			List<CanvasItem *> selection = _get_edited_canvas_items();
+
+			drag_selection.clear();
+			for (int i = 0; i < selection.size(); i++) {
```
Comment (KoBeWi) [inline editor/plugins/canvas_item_editor_plugin.cpp:2104]:
```suggestion
			for (CanvasItem *E : selection) {
```
Using range iterators is better when possible.

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Could you squash the two commits together? See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase).

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 2]
```diff
@@ -2097,8 +2097,16 @@ bool CanvasItemEditor::_gui_input_move(const Ref<InputEvent> &p_event) {
 	if (k.is_valid() && k->is_pressed() && (tool == TOOL_SELECT || tool == TOOL_MOVE) &&
 			(k->get_keycode() == Key::UP || k->get_keycode() == Key::DOWN || k->get_keycode() == Key::LEFT || k->get_keycode() == Key::RIGHT)) {
 		if (!k->is_echo()) {
-			// Start moving the canvas items with the keyboard
-			drag_selection = _get_edited_canvas_items();
+			// Start moving the canvas items with the keyboard, if they are movable
+			List<CanvasItem *> selection = _get_edited_canvas_items();
+
+			drag_selection.clear();
+			for (CanvasItem* item : selection) {
```
[the remark this answers is not in the document]
Comment (LlamaLad7) [reply editor/plugins/canvas_item_editor_plugin.cpp:2104]: Oops, will fix

**Comment by LlamaLad7** [comment]
> Could you squash the two commits together? See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase).

My bad I thought they'd be squashed when merging

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by Calinou** [comment]
> My bad I thought they'd be squashed when merging

We don't use GitHub's squash and merge feature on the main Godot repository, since it loses the merge commit in the process. We prefer keeping the merge commit to make the Git history more self-documented (even if that makes it longer when not using the `--no-merges` option).

**Comment by LlamaLad7** [comment]
Got it. Thanks for the info.

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Pull request merged**